tests/unit/dccs/test_aws_iot.py

```diff
@@ -0,0 +1,615 @@
+        # Pop metric_data key from dict
+        d1.pop("metric_data", None)
+        d2.pop("metric_data", None)
+
+        for k, v1 in d1.iteritems():
+            # Check if key present in the second dict
+            self.assertIn(k, d2, msg)
+
+            # Get value of key from second dict
+            v2 = d2[k]
+
+            # Compare the value
+            self.assertEqual(v1, v2, msg)
+
+        return True
+
+    def test_aws_class_implementation(self):
+        """
+        Method to test the implementation of AWSIoT class.
+        :return: None
+        """
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        self.assertIsInstance(aws, AWSIoT, "Check the implementation of AWSIoT class")
+
+    def test_validation_of_comms_parameter(self):
+        """
+        Method to test the validation of AWSIoT class for invalid connections object.
+        :return: None
+        """
+        # Checking whether implementation raising the TypeError Exception invalid comms object
+        with self.assertRaises(Exception):
+            AWSIoT("Invalid object", enclose_metadata=True)
+
+    def test_implementation_register_entity(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Register the edge
+        registered_entity = aws.register(self.edge_system)
+
+        # Check the returned object is of the class RegisteredEntity
+        self.assertIsInstance(registered_entity, RegisteredEntity)
+
+    def test_implementation_register_metric(self):
+        """
+        Method to test the implementation of register method of AWSIoT for entity type.
+        :return: None
+        """
+
+        # Monkey patch the constructor of MqttDccComms
+        MqttDccComms.__init__ = mocked_init_mqtt_dcc_comms
+
+        # Creating the mocked MqttDccComms object
+        mocked_mqtt_dcc_comms = MqttDccComms()
+
+        aws = AWSIoT(mocked_mqtt_dcc_comms, enclose_metadata=True)
+
+        # Creating test Metric
+        test_metric = Metric(
+            name="Test_Metric",
+            unit=None,
+            interval=10,
+            aggregation_size=2,
+            sampling_function=sampling_function
+        )
+
+        registered_metric = aws.register(test_metric)
+
+        # Check the returned object is of the class RegisteredMetric
+        self.assertIsInstance(registered_metric, RegisteredMetric)
+
+    def test_implementation_create_relationship(self):
+        """
```

Comment:
Change comment to:
"Test case to check RegisteredEntity and Parent and RegisteredMetric as child"